test/unit/statistics_test.rb

```diff
@@ -0,0 +1,41 @@
+require 'test_helper'
+
+class StatisticsTest < ActiveSupport::TestCase
+  def setup
+   @user = users(:bob)
+  end
+
+  test 'daily_note_tally returns the correct type of array' do
+      daily = @user.daily_note_tally
+      assert_not_empty daily
+      assert_equal daily, 365
+  end
+
+  test 'weekly_note_tally returns the correct type of array' do
+    weekly = @user.weekly_note_tally
+    assert_equal weekly.count, 53
+    assert_not_nil weekly
+  end
+
+test 'note_streak' do
+  assert_not_nil @user.note_streak.count
+end
+
+test 'wiki_edit streak' do
+  assert_not_nil @user.wiki_edit_streak
+end
+
+test 'weekly_comment_tally' do
+  assert_not_nil @user.weekly_comment_tally
```

Comment:
For things like this, we could confirm the /format/ of the returned items; we should leave the values untested because as people change our fixtures in the future, it'll cause side-effects by breaking these tests. But we can confirm that it returns, for example, 52 items, and maybe something about the parameter names of each item? Is there a Ruby equivalent of JS's `Object.keys()` which returns the key names of an object or associative array? 

Thanks, Cess! Great future-proofing!

Reply:
Thanks @jywarren :smiley: I will implement the count assertions and  research how to assert a hash format was returned or something.